New CI test "CI / Build docs (pull_request)" fails

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**System information**

- Cantera version: development branch
- OS: N/A - GitHub CI
- Python/MATLAB version: N/A

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->
CI tests run successfully if `scons test` and `scons build` pass.

**Actual behavior**

<!-- A clear and concise description of what the bug is. -->
The **CI / Build docs (pull_request)** test errors:
```
Set up SSH key and host for deploy
1s
##[error]Input required and not supplied: key
Run shimataro/ssh-key-action@v2
##[error]Input required and not supplied: key
```

**Additional context**

<!-- Add any other context about the problem here. -->
See failed test at: #881

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -138,6 +138,7 @@ jobs:
       # machine that has previously logged in to cantera.org and trusts
       # that it logged in to the right machine
       - name: Set up SSH key and host for deploy
+        if: github.event_name == 'push' && github.repository_owner == 'Cantera' && github.ref == 'master'
         uses: shimataro/ssh-key-action@v2
         with:
           key: ${{ secrets.CTDEPLOY_KEY }}

